src/Features/CSharp/Portable/Completion/CompletionProviders/PropertySubPatternCompletionProvider.cs

```diff
@@ -0,0 +1,177 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Completion;
+using Microsoft.CodeAnalysis.Completion.Providers;
+using Microsoft.CodeAnalysis.CSharp.Extensions;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Options;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.CodeAnalysis.CSharp.Completion.Providers
+{
+    internal class PropertySubPatternCompletionProvider : CommonCompletionProvider
+    {
+        public override async Task ProvideCompletionsAsync(CompletionContext context)
+        {
+            var document = context.Document;
+            var position = context.Position;
+            var cancellationToken = context.CancellationToken;
+
+            var workspace = document.Project.Solution.Workspace;
+            var semanticModel = await document.GetSemanticModelForSpanAsync(new TextSpan(position, length: 0), cancellationToken).ConfigureAwait(false);
+            var (type, location) = GetPatternType(document, semanticModel, position, cancellationToken);
+
+            if (type == null)
+            {
+                return;
+            }
+
+            // Find the members that can be tested.
+            IEnumerable<ISymbol> members = semanticModel.LookupSymbols(position, type);
+            members = members.Where(m => m.CanBeReferencedByName &&
+                (m.IsKind(SymbolKind.Property) || m.IsKind(SymbolKind.Field)) &&
+                !m.IsImplicitlyDeclared);
+
+            // Filter out those members that have already been typed
+            var alreadyTestedMembers = GetTestedMembers(semanticModel.SyntaxTree, position, cancellationToken);
+            var untestedMembers = members.Where(m => !alreadyTestedMembers.Contains(m.Name));
+
+            untestedMembers = untestedMembers.Where(m => m.IsEditorBrowsable(document.ShouldHideAdvancedMembers(), semanticModel.Compilation));
+
+            foreach (var untestedProperty in untestedMembers)
+            {
+                context.AddItem(SymbolCompletionItem.CreateWithSymbolId(
+                    displayText: untestedProperty.Name,
+                    insertionText: null,
```

Comment:
is this correct?  what happens if hte property has a keyword as a name?

Reply:
Fixed